crypto/dsa/dsa_check.c

```diff
@@ -19,14 +19,25 @@
 #include "dsa_local.h"
 #include "crypto/dsa.h"
 
-int dsa_check_params(const DSA *dsa, int *ret)
+int dsa_check_params(const DSA *dsa, int *codes)
 {
-    /*
-     * (2b) FFC domain params conform to FIPS-186-4 explicit domain param
-     * validity tests.
-     */
-    return ossl_ffc_params_FIPS186_4_validate(dsa->libctx, &dsa->params,
-                                              FFC_PARAM_TYPE_DSA, ret, NULL);
+    FFC_PARAMS params = {0};
+    int ret;
+
+    if (dsa->params.seed != NULL)
+        /*
+         * (2b) FFC domain params conform to FIPS-186-4 explicit domain param
+         * validity tests.
+         */
+        return ossl_ffc_params_FIPS186_4_validate(dsa->libctx, &dsa->params,
+                                                  FFC_PARAM_TYPE_DSA, codes, NULL);
+
+    if (!ossl_ffc_params_copy(&params, &dsa->params))
```

Comment:
Why is the copy here?

Reply:
Because the dsa->params are const and we need to change the flags to do the validation of g.